test/cri-containerd/containerdrestart_test.go

```diff
@@ -68,3 +71,296 @@ func Test_ContainerdRestart_LCOW(t *testing.T) {
+			GetStateCommand: []string{"cmd", "/c", "type", testFile},
+			ExpectedResult:  "- \r\n",
+		},
+	}
+
+	for _, tt := range tests {
+		for _, restart := range []bool{false, true} {
+			suffix := "_Restart"
+			if !restart {
+				suffix = "_No" + suffix
+			}
+
+			t.Run(tt.Name+suffix, func(t *testing.T) {
+				requireFeatures(t, tt.Feature)
+				if restart {
+					requireFeatures(t, featureTerminateOnRestart)
+				}
+
+				sandboxRequest := getRunPodSandboxRequest(t, tt.Runtime,
+					append(tt.SandboxOpts,
+						WithSandboxAnnotations(map[string]string{
+							criAnnotations.EnableReset: "true",
+						}))...)
+
+				podID := runPodSandbox(t, client, ctx, sandboxRequest)
+				defer removePodSandbox(t, client, ctx, podID)
+				defer stopPodSandbox(t, client, ctx, podID)
+
+				request := getCreateContainerRequest(podID, t.Name()+"-Container", tt.Image, tt.Command, sandboxRequest.Config)
+				request.Config.Annotations = map[string]string{
+					criAnnotations.EnableReset: "true",
+				}
+
+				containerID := createContainer(t, client, ctx, request)
+				startContainer(t, client, ctx, containerID)
+				defer removeContainer(t, client, ctx, containerID)
+				defer func() {
+					stopContainer(t, client, ctx, containerID)
+				}()
+
+				execRequest := &runtime.ExecSyncRequest{
+					ContainerId: containerID,
+					Cmd:         tt.SetStateCommand,
+					Timeout:     1,
+				}
+				req := execSync(t, client, ctx, execRequest)
+				if req.ExitCode != 0 {
+					t.Fatalf("exec %v failed with exit code %d: %s", execRequest.Cmd, req.ExitCode, string(req.Stderr))
+				}
+
+				// check the write worked
+				execRequest = &runtime.ExecSyncRequest{
+					ContainerId: containerID,
+					Cmd:         tt.GetStateCommand,
+					Timeout:     1,
+				}
+
+				req = execSync(t, client, ctx, execRequest)
+				if req.ExitCode != 0 {
+					t.Fatalf("exec %v failed with exit code %d: %s %s", execRequest.Cmd, req.ExitCode, string(req.Stdout), string(req.Stderr))
+				}
+
+				if string(req.Stdout) != tt.ExpectedResult {
+					t.Fatalf("did not properly set container state; expected %q, got: %q", tt.ExpectedResult, string(req.Stdout))
+				}
+
+				/*******************************************************************
+				* restart pod
+				*******************************************************************/
+				stopContainer(t, client, ctx, containerID)
+				stopPodSandbox(t, client, ctx, podID)
+
+				if restart {
+					// allow for any garbage collection and clean up to happen
+					time.Sleep(time.Second * 1)
+					stopContainerd(t)
+					startContainerd(t)
+				}
+
+				newPodID := runPodSandbox(t, client, ctx, sandboxRequest)
+				if newPodID != podID {
+					defer removePodSandbox(t, client, ctx, newPodID)
+					defer stopPodSandbox(t, client, ctx, newPodID)
+					t.Fatalf("pod restarted with different id (%q) from original (%q)", newPodID, podID)
+				}
+
+				startContainer(t, client, ctx, containerID)
+
+				execRequest = &runtime.ExecSyncRequest{
```

Comment:
nit: no need to reinitialize? maybe also rename to `stateExecRequest` or something.